tools/restatectl/src/commands/node/list_nodes.rs

```diff
@@ -77,9 +97,90 @@ pub async fn list_nodes(connection: &ConnectionInfo, _opts: &ListNodesOpts) -> a
             ),
         ];
+
+        if opts.extra {
+            node_row.extend(render_extra_columns(
+                nodes_extra_info.get(&node_id),
+                |ident_response| {
+                    vec![
+                        duration_to_human_rough(
+                            TimeDelta::seconds(ident_response.age_s as i64),
+                            Tense::Present,
+                        ),
+                        // We reuse the prost-generated Debug formatting but cut out the "Unknown"s
+                        format!("{:?}", ident_response.status()).replace("Unknown", "-"),
+                        format!("{:?}", ident_response.admin_status()).replace("Unknown", "-"),
+                        format!("{:?}", ident_response.worker_status()).replace("Unknown", "-"),
+                        format!("{:?}", ident_response.log_server_status()).replace("Unknown", "-"),
+                        format!("{:?}", ident_response.metadata_server_status())
+                            .replace("Unknown", "-"),
+                        format!("{}", ident_response.nodes_config_version),
+                        format!("{}", ident_response.logs_version),
+                        format!("{}", ident_response.schema_version),
+                        format!("{}", ident_response.partition_table_version),
+                    ]
+                    .iter()
+                    .map(Cell::new)
+                    .collect()
+                },
+            ));
+        }
         nodes_table.add_row(node_row);
     }
     c_println!("{}", nodes_table);
 
     Ok(())
 }
+
+fn render_extra_columns<F>(ident_response: Option<&IdentResponse>, f: F) -> Vec<Cell>
+where
+    F: FnOnce(&IdentResponse) -> Vec<Cell>,
+{
+    ident_response
+        .map(f)
+        .unwrap_or(vec![Cell::new("N/A"), Cell::new("Unknown")])
+}
+
+async fn fetch_extra_info(
+    nodes_configuration: &NodesConfiguration,
+) -> anyhow::Result<HashMap<PlainNodeId, IdentResponse>> {
+    let mut get_ident_tasks = JoinSet::<anyhow::Result<IdentResponse>>::new();
+
+    for (node_id, node_config) in nodes_configuration.iter() {
+        let address = node_config.address.clone();
+        let get_ident = async move {
+            let node_channel = grpc_connect(address).await?;
+            let mut node_svc_client =
+                NodeSvcClient::new(node_channel).accept_compressed(CompressionEncoding::Gzip);
+
+            Ok(node_svc_client
+                .get_ident(())
+                .await
+                .map_err(|e| {
+                    // we ignore individual errors in table rendering so this is the only place to log them
+                    c_println!("failed to call GetIdent for node {}: {:?}", node_id, e);
+                    anyhow::anyhow!(e)
+                })?
+                .into_inner())
+        };
+        get_ident_tasks.spawn(async move {
+            match tokio::time::timeout(std::time::Duration::from_secs(1), get_ident).await {
```

Comment:
I suggest we define a constant for the timeout so it can be better documented and modified if needed

Reply:
Ack, will do before merge. Generally I try keep these as close to the place they're used as possible; constants are useful if shared but for something like this, it makes less sense. A pattern I've seen used is to ship a set of config along with the client of a given service that prescribes that the service expects a reasonable timeout should be. Perhaps it's something we can explore across the runtime.